pkg/actors/placement/placement.go

```diff
@@ -288,23 +294,40 @@ func (p *actorPlacement) WaitUntilReady(ctx context.Context) error {
 }
 
 // LookupActor resolves to actor service instance address using consistent hashing table.
-func (p *actorPlacement) LookupActor(actorType, actorID string) (string, string) {
+func (p *actorPlacement) LookupActor(ctx context.Context, req internal.LookupActorRequest) (internal.LookupActorResponse, error) {
+	// Retry here to allow placement table dissemination/rebalancing to happen.
+	policyDef := p.resiliency.BuiltInPolicy(resiliency.BuiltInActorNotFoundRetries)
+	policyRunner := resiliency.NewRunner[internal.LookupActorResponse](ctx, policyDef)
+	return policyRunner(func(ctx context.Context) (res internal.LookupActorResponse, rErr error) {
+		rAddr, rAppID, rErr := p.doLookupActor(ctx, req.ActorType, req.ActorID)
+		if rErr != nil {
+			return res, fmt.Errorf("error finding address for actor %s/%s: %w", req.ActorType, req.ActorID, rErr)
+		} else if rAddr == "" {
+			return res, fmt.Errorf("did not find address for actor %s/%s", req.ActorType, req.ActorID)
+		}
+		res.Address = rAddr
+		res.AppID = rAppID
+		return res, nil
+	})
+}
+
+func (p *actorPlacement) doLookupActor(ctx context.Context, actorType, actorID string) (string, string, error) {
 	p.placementTableLock.RLock()
 	defer p.placementTableLock.RUnlock()
 
 	if p.placementTables == nil {
-		return "", ""
+		return "", "", errors.New("placement tables are not set")
 	}
 
 	t := p.placementTables.Entries[actorType]
 	if t == nil {
-		return "", ""
+		return "", "", nil
 	}
 	host, err := t.GetHost(actorID)
 	if err != nil || host == nil {
-		return "", ""
+		return "", "", nil //nolint:nilerr
```

Comment:
Any reason we don't check for these separately and log + return the error if there is one?

Reply:
This is what the previous code was doing. I am letting the empty address "bubble up" so the caller can eal with it